docs/dub-reference/subpackages.md

```diff
@@ -6,7 +6,90 @@
 
     Tracking issue: <https://github.com/dlang/dub-docs/issues/71>
 
-For adding sub-packages from the registry as a dependency, execute `dub add <packagename>:<subpackage>`
+## Introduction
 
-If you need to compile a sub-package only, use `dub build :<subpackage>`
+DUB has support for sub-packages, a mechanic used to define additional optional
+recipes within the main recipe file. They can be built and tested separately.
 
+The main distinction to configurations is the intent of use. For example,
+while configurations are typically used to separate server and client code,
+sub-packages can be used to create and manage dependencies used by
+these server and client example configurations.
+
+Another distinction with configurations, if at least one configuration is defined,
+the first configuration is taken as default. This does not happen with sub-packages.
+
+Each sub-package can have its own set of configurations and build options.
+
+## Usage
+
+The syntax used to reference sub-packages in commands is often `<package>:<sub-package>` or just `:<sub-package>`:
+- To only add this sub-package from a package as a dependency, execute `dub add <packagename>:<sub-package>`
+- To build a sub-package, execute `dub build :<sub-package>`
+- To build and run a sub-package, execute `dub run :<sub-package>`
+- To test a sub-package, execute `dub test :<sub-package>`
+
+When the package name is omitted, the current working package name is assumed.
+
+The default `targetType` for sub-packages is `library`.
+
+## Declaration
+
+To be useful, a sub-package needs, at minimum, a name, import paths,
+and source paths (or source files).
```

Comment:
import paths and source paths don't have to be set, they work the same as in a regular package.